### ISSUE TITLE
[FEATURE REQUEST] TWO-SHOT Drawing masks (from Latent couple)

### ISSUE THREAD

**Issue by FemBoxbrawl** [Haoming02/sd-forge-couple#33]
I was wondering if you could add an option to also use color coded masks (like in the old Stable-diffusion two shot), where each color is a specific area that you prompt.

The thing is, the current Latent couple extension does not work with forge. SO I was wondering if you could also add it to Forge couple?
The reason i am asking is because with the tiles, you are limited to what goes wear, while with the color masking, you can have seperated areas as a mask (lets say some complex fight scene where the head and arm is visible but the torso is not, and then another guy is blocking that view, so you'd have to somehow mask it so that prompt belong to the same one.

Basically a color coded masks that can be "seperate" but would still follow the prompt.

**Comment by Haoming02** [comment]
Not sure if I follow

Doesn't the `Advanced` mapping already do this?

**Comment by FemBoxbrawl** [comment]
> Not sure if I follow
> 
> Doesn't the `Advanced` mapping already do this?

You are right, the tile does work to some extent, . But I meant like a "masking" option, where you can draw over specific areas, so for example 2 seperate "regions" for the same line/prompt. (to be able to make Overlaps, etc.

**Comment by rktvr** [comment]
> Not sure if I follow
> 
> Doesn't the `Advanced` mapping already do this?

they mean a mask you can draw, for even more control

**Comment by rktvr** [comment]
any plans for being able to draw a mask?

**Comment by Haoming02** [comment]
After I finish revamping `Advanced` mode to support the new Forge I guess

Still need a good way to store multiple "layers" of images after you draw the mask though...

**Comment by Haoming02** [comment]
Aight, time to think of a way to implement `Mask` mode to the UI...

**Comment by queenofinvidia** [comment]
Awesome, can't wait for the update! Does Forge Couple work on new Forge now? I wasn't sure if I need to reinstall it from here or wait for the integrated version inside of Forge to be updated.

I'm SUPER excited for masking mode as this was pretty much the only feature Regional Prompter had that Forge Couple did not.

Thank you so much for all your work!

**Comment by Haoming02** [comment]
> Does Forge Couple work on new Forge now? 

Yes. It should work for the new Forge now. *(Only `SD 1.5` and `SDXL`, but not `Flux`)*

**Comment by Haoming02** [comment]
~~*Slowly going insane from working with Gradio 3 `Image` component...*~~

**Comment by rktvr** [comment]
> Hello, please give the [mask](https://github.com/Haoming02/sd-forge-couple/tree/mask) branch a try~
> 
> _Only works for old Forge (`Gradio 3`) right now..._

playing around with it at the moment, needs a way to clear the actual canvas. when i "Save mask" it doesn't clear the canvas, so it ends up duplicating everything and i don't see where to clear it.

**Comment by rktvr** [comment]
> Oh, click the `Create Empty Canvas` button to clear the canvas.

doesn't work, when i click it the anvas stasy the same. is probably because of reForge though.

**Comment by Haoming02** [comment]
> doesn't work, when i click it the anvas stasy the same. is probably because of reForge though

Wait, really? Let me try.

**Comment by rktvr** [comment]
i'll assume that any errors are because of the (somewhat?) old version of reForge i'm using. I'll see if anything changes if I update it though I don't remember if they started using gradio 4 in the newer reForge.

**Comment by Haoming02** [comment]
Huh, it's working for me on reForge though. Using the `main` branch.

Let me try `dev_upstream` branch real quick.

**Comment by rktvr** [comment]
apparently this is the version i'm on: https://github.com/Panchovix/stable-diffusion-webui-reForge/commit/a75e69cb3528726e11a41f23fdec745a9f03d795

I'll update and re-check everything again. it was likely just because of that version of reforge

**Comment by Haoming02** [comment]
Clicking the `Create Empty Canvas` button empties the canvas for me, on both `main` and `dev_upstream` branch of reForge.

**Comment by rktvr** [comment]
Just updated and it does indeed clear now! Thank you for cour continued work and support on this add-on! 🥳

**Comment by queenofinvidia** [comment]
awesome! Thank you! I'm gonna download this and try it out this week. will post here with my experience <3

**Comment by queenofinvidia** [comment]
sorry if this is a silly question but how do I update to get the new features with adding a background image and masking feature?

I updated on Forge, but I'm still on version 2.0.3 and don't see any new features.

I made an issue page too, I'm not sure if that makes it easier. https://github.com/Haoming02/sd-forge-couple/issues/54

thank you for your help! I really would love to try the new features ^^

**Comment by metapea** [comment]
There was code added code to OG forge for Latent couple

> It is a bit late but "online loras" are currently available now: https://github.com/lllyasviel/stable-diffusion-webui-forge/blob/2f1d04759f36122cbb3553f0f0758755790624c7/backend/operations.py#L14 - lllyasviel

>  View “online lora” as a module similar to ControlLoRA so that it is moved to GPU together with model when sampling, achieving significant speedup and perfect low VRAM management simultaneously. - lllyasviel
